azure/packages/azure-client/src/AzureClient.ts

```diff
@@ -303,5 +305,15 @@ export class AzureClient {
 		fluidContainer.attach = attach;
 		return fluidContainer;
 	}
+
+	private async getContainerEntryPoint(container: IContainer): Promise<IRootDataObject> {
+		const rootDataObject: FluidObject<IRootDataObject> | undefined =
+			await container.getEntryPoint();
+		assert(
+			rootDataObject?.IRootDataObject !== undefined,
+			"entryPoint must be of type IRootDataObject",
+		);
+		return rootDataObject.IRootDataObject;
```

Comment:
@anthony-murphy @kian-thompson This is establishing a new contract across the loader-runtime boundary, which will make old runtimes (fluid-static) incompatible with this new loader (azure-client).

I think we are already planning to require this restriction based on the characteristics of the cross-version compat testing we're adding, but I want to make sure we're explicitly acknowledging this.

Reply:
We support older (LTS) versions of loader but not runtime for which we support N-1. Basically, new loader (say N) cannot be used with runtimes older than N-1 AFAIK.